src/evaluators/CallExpression.js

```diff
@@ -302,15 +300,39 @@ function EvaluateCall(
 ): Value {
   if (func instanceof AbstractValue) {
     let loc = ast.callee.type === "MemberExpression" ? ast.callee.property.loc : ast.callee.loc;
+    if (func.kind === "conditional") {
+      let [condValue, consequentVal, alternateVal] = func.args;
+      invariant(condValue instanceof AbstractValue);
+      if (
+        Value.isTypeCompatibleWith(consequentVal.getType(), FunctionValue) ||
+        Value.isTypeCompatibleWith(alternateVal.getType(), FunctionValue)
+      ) {
+        return callBothFunctionsAndJoinTheirEffects(
+          condValue,
+          consequentVal,
+          alternateVal,
+          ast,
+          strictCode,
+          env,
+          realm
+        );
+      }
+    } else if (func.kind === "||") {
+      let [leftValue, rightValue] = func.args;
+      invariant(leftValue instanceof AbstractValue);
```

Comment:
Why are you not checking if leftValue and rightValue are known to be functions? Alternatively, why are you checking if consequentVal and alternateVal are functions?

Reply:
I've added the same checks as for the conditional above. We only check if either of the values are function values, otherwise we don't bother trying to optimize as we'll just end up producing more output without gaining any precision.